src/components/FormElements/AssetPicker.tsx

```diff
@@ -1,271 +1,145 @@
-import React, { useState } from "react";
+import React from "react";
```

Comment:
I simplified the `AssetPicker` component always to accept and return an object. Then, the view can handle the response as it fits that view (string or object).

The validation was also moved outside of the form elements to make them more reusable.